src/app/space/add-space-overlay/add-space-overlay.component.ts

```diff
@@ -0,0 +1,140 @@
+import { UserService } from 'ngx-login-client';
+import { Observable } from 'rxjs';
+
+import { ContextService } from 'app/shared/context.service';
+import { DummyService } from 'app/shared/dummy.service';
+import { SpaceNamespaceService } from 'app/shared/runtime-console/space-namespace.service';
+import { SpacesService } from 'app/shared/spaces.service';
+
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'f8-add-space-overlay',
+  styleUrls: ['./add-space-overlay.component.less'],
+  templateUrl: './add-space-overlay.component.html'
+})
+export class AddSpaceOverlayComponent implements OnInit {
+  currentSpace: Space;
+  selectedTemplate: ProcessTemplate;
+  spaceTemplates: ProcessTemplate[];
+  space: Space;
+
+  constructor(private router: Router,
+              public dummy: DummyService,
+              private spaceService: SpaceService,
+              private notifications: Notifications,
+              private userService: UserService,
+              private spaceNamespaceService: SpaceNamespaceService,
+              private spaceNamePipe: SpaceNamePipe,
+              private spacesService: SpacesService,
+              private context: ContextService,
+              private broadcaster: Broadcaster) {
+    this.spaceTemplates = dummy.processTemplates;
+    this.space = this.createTransientSpace();
+  }
+
+  ngOnInit() {
+    const srumTemplates = this.spaceTemplates.filter(template => template.name === 'Scenario Driven Planning');
+    if (srumTemplates && srumTemplates.length > 0) {
+      this.selectedTemplate = srumTemplates[0];
+    }
+    this.context.current.subscribe((ctx: Context) => {
+      if (ctx.space) {
+        this.currentSpace = ctx.space;
+      }
+    });
+  }
+
+  /*
+   * Creates a persistent collaboration space
+   * by invoking the spaceService
+   */
+  createSpace() {
+    console.log('Creating space', this.space);
+    if (!this.space) {
+      this.space = this.createTransientSpace();
+    }
+    this.space.attributes.name = this.space.name.replace(/ /g, '_');
+    this.userService.getUser()
+      .switchMap(user => {
+        this.space.relationships['owned-by'].data.id = user.id;
+        return this.spaceService.create(this.space);
+      })
+      .do(createdSpace => {
+        this.spacesService.addRecent.next(createdSpace);
+      })
+      .switchMap(createdSpace => {
+        return this.spaceNamespaceService
+          .updateConfigMap(Observable.of(createdSpace))
+          .map(() => createdSpace)
+          // Ignore any errors coming out here, we've logged and notified them earlier
+          .catch(err => Observable.of(createdSpace));
+      })
+      .subscribe(createdSpace => {
+          this.router.navigate([createdSpace.relationalData.creator.attributes.username,
+            createdSpace.attributes.name]);
+          this.showAddAppOverlay();
+          this.hideAddSpaceOverlay();
+        },
+        err => {
+          console.log('Error creating space', err);
```

Comment:
Please add a call to the global error handler so that this is picked up by Sentry.

Reply:
These should just be rethrown unless there is something the user can do.